src/clr-angular/timeline/timeline-step.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, Input } from '@angular/core';
+
+import { ClrTimelineStepState } from './timeline-step-state.enum';
+
+interface IconAttribute {
+  iconShape: string | undefined;
+  ariaLabel: string;
+}
+
```

Comment:
I have one request here it is related to the `ClrCommonStrings`. We are not providing any way to let the user change the `ariaLabel` or what is the message inside the `ClrSpinner`.

```ts
const ICON_ATTR_MAP: Record<ClrTimelineStepState, IconAttribute> = {
  [ClrTimelineStepState.NOT_STARTED]: { iconShape: 'circle', ariaLabel: commonStrings.keys.timlineStateNotStarted },
  [ClrTimelineStepState.CURRENT]: { iconShape: 'dot-circle', ariaLabel: commonString.keys.timelineStateCurrent },
  // ...
```

Also with the planned deprecation of the `ClrAriaLiveService` in version 4, we gonna remove the service from the `ClrSpinner` so the text between the open & closing tag the text will become obsolete. Right now the text is used to be passed to the `aria-live` service but this won't be the case in the next version. So I see two options here:
  
 * Not providing translation for `Fetching data` or even not adding it to the template.
 * Provide translation now, but deprecate it, so it could be removed in version 4 as well as the service.

Currently, SR will announce `Fetching data` when the `#processing` template is visible. But will create dependency when  we made some changes to the `ClrSpinner`